openedx/tests/xblock_integration/test_external_xblocks.py

```diff
@@ -0,0 +1,37 @@
+"""
+This will run tests on all XBlocks in the `xblock.test.v0`
+entrypoint. Did you notice something about that entry point? It ends
+with a v0. That means this is not finished. At some point, we might
+stop running v0 tests, replacing them with test case failures, and
+run v1 tests only.
+
+That be the dragon here.
+"""
+import pkg_resources
+
+class DuplicateXBlockTest(Exception):
+    pass
+
+class InvalidTestName(Exception):
+    pass
+
+loaded = False
+
+for entrypoint in pkg_resources.iter_entry_points(group="xblock.test.v0"):
+    plugin = entrypoint.load()
+    classname = str(entrypoint).split(":")[-1]
```

Comment:
Thank you. I will change to plugin.__name__.

I would like to keep the validity checks, however. For better or worse, the Python test frameworks do rely on naming to figure out which things are tests, and I'd like to enforce such conventions here. Even as a sanity check, I want to catch errors like: 

* Someone makes an entrypoint that points to a non-test class. E.g. they cut-and-paste in setup.py, and fail to update one field, and point to the XBlock itself, or any of a range of similar errors. 
* There are two versions of the same XBlock installed, or two XBlocks have a name collision.

It's also good security practice. Without such checks, we're giving XBlock authors freedom to inject anything into globals() here otherwise. I agree that this doesn't raise any security issues /right now/, but checks like this make it much less likely we run into security issues as the code evolves in the future.